src/test/java/net/dv8tion/jda/test/modals/ModalTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.test.modals;
+
+import net.dv8tion.jda.api.components.Component;
+import net.dv8tion.jda.api.components.attribute.IDisableable;
+import net.dv8tion.jda.api.components.label.Label;
+import net.dv8tion.jda.api.components.label.LabelChildComponent;
+import net.dv8tion.jda.api.components.selections.EntitySelectMenu;
+import net.dv8tion.jda.api.components.selections.StringSelectMenu;
+import net.dv8tion.jda.api.modals.Modal;
+import net.dv8tion.jda.test.components.ComponentTestData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+public class ModalTest {
+    static Map<Component.Type, Class<? extends LabelChildComponent>> LABEL_CHILDREN = Map.of(
+            Component.Type.STRING_SELECT, StringSelectMenu.class,
+            Component.Type.USER_SELECT, EntitySelectMenu.class,
+            Component.Type.ROLE_SELECT, EntitySelectMenu.class,
+            Component.Type.MENTIONABLE_SELECT, EntitySelectMenu.class,
+            Component.Type.CHANNEL_SELECT, EntitySelectMenu.class);
+
+    @MethodSource("modalComponents")
+    @ParameterizedTest
+    public void testDisabledComponents(Component component) {
+        Assertions.assertInstanceOf(IDisableable.class, component);
+        Assertions.assertInstanceOf(LabelChildComponent.class, component);
+        LabelChildComponent mutatedComponent = ((LabelChildComponent) ((IDisableable) component).asDisabled());
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            Modal.create("id", "title")
+                    .addComponents(Label.of("label", mutatedComponent))
+                    .build();
+        });
+    }
+
+    @MethodSource("modalComponents")
+    @ParameterizedTest
+    public void testEnabledComponents(Component component) {
```

Comment:
You can directly put `LabelChildComponent component` as the test parameter, since JUnit will already check the type for you. This removes the need for the instanceof check.

Reply:
This lets the test input mirror the signature of `Modal#componentIsEnabled`, and I consider the `LabelChildComponent` check part of the test itself, so I'd rather have it fail in the test body than in the input. 

Ideally I would have tested `componentIsEnabled` directly (And had it be package-private), but since the Test package path is `net.dv8tion.jda.test` I'd have to put the Test in an awkward package from the rest of the test.

Reply:
The reason I prefer changing it, is because it makes the test more explicit, since you're actually testing `LabelChildComponent` and not just `Component`. The test additionally wraps all its input in a `Label`, which you can only know if you actually read the test code. Alternatively, you could rename the test to `testEnabledLabelChildComponent(Component component)` but that ends up achieving the same thing as just changing the parameter type, with the downside of still having to explicitly add an assertion.

I think it's good to test the modal's validation through the modal builder, instead of only testing the internal check method.